pkg/ctl/topic/set_retention.go

```diff
@@ -0,0 +1,197 @@
+	}
+
+	tenantNotExistError := cmdutils.Output{
+		Desc: "the tenant does not exist",
+		Out:  "[✖]  code: 404 reason: Tenant does not exist",
+	}
+
+	nsNotExistError := cmdutils.Output{
+		Desc: "the namespace does not exist",
+		Out:  "[✖]  code: 404 reason: Namespace (tenant/namespace) does not exist",
+	}
+
+	out = append(out, successOut, noTopicName, tenantNotExistError, nsNotExistError)
+	desc.CommandOutput = out
+
+	vc.SetDescription(
+		"set-retention",
+		"Set the retention policy for a topic",
+		desc.ToString(),
+		desc.ExampleToString(),
+		"set-retention",
+	)
+
+	var timeStr string
+	var sizeStr string
+
+	vc.Command.Flags().StringVarP(&timeStr, "time", "", "",
+		"Retention time in minutes (or minutes, hours, days, weeks eg: 100m, 3h, 2d, 5w). "+
+			"0 means no retention and -1 means infinite time retention")
+	vc.Command.Flags().StringVarP(&sizeStr, "size", "", "",
+		"Retention size limit (eg: 10M, 16G, 3T). "+
+			"0 or less than 1MB means no retention and -1 means infinite size retention")
+
+	vc.SetRunFuncWithNameArg(func() error {
+		return doSetRetention(vc, timeStr, sizeStr)
+	}, "the topic name is not specified or the topic name is specified more than one")
+}
+
+func doSetRetention(vc *cmdutils.VerbCmd, timeStr string, sizeStr string) error {
+	topic := vc.NameArg
+
+	if timeStr == "" {
+		return errors.New("missing option: --time")
+	}
+	if sizeStr == "" {
+		return errors.New("missing option: --size")
+	}
+
+	duration, err := parseRetentionTimeString(timeStr)
+	if err != nil {
+		return err
+	}
+
+	size, err := parseRetentionSizeString(sizeStr)
+	if err != nil {
+		return err
+	}
+
+	admin := cmdutils.NewPulsarClient()
+	topicName, err := util.GetTopicName(topic)
+	if err != nil {
+		return err
+	}
+
+	err = admin.Topics().SetRetention(*topicName, util.RetentionData{
+		RetentionSizeInMB:      size,
+		RetentionTimeInMinutes: duration,
+	})
+	if err == nil {
+		vc.Command.Printf("Set the retention policy successfully on [%s]\n", topic)
+	}
+
+	return err
+}
+
+func parseRetentionSizeString(size string) (int64, error) {
+	if size == "" {
+		return 0, errors.New("size cannot be empty")
+	}
+
+	sizeUnit := strings.ToUpper(size[len(size)-1:])
+	if unicode.IsNumber([]rune(sizeUnit)[0]) {
+		return strconv.ParseInt(size, 10, 64)
+	}
+
+	value, err := strconv.ParseInt(size[0:len(size)-1], 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	switch sizeUnit {
+	case "K":
+		return value * 1024, nil
+	case "M":
+		return value * 1024 * 1024, nil
+	case "G":
+		return value * 1024 * 1024 * 1024, nil
+	case "T":
+		return value * 1024 * 1024 * 1024 * 1024, nil
+
+	default:
+		return 0, fmt.Errorf("cannot parse %s to number format size", size)
+	}
+}
+
+var timeMinute = time.Minute.Seconds()
+var timeHour = time.Hour.Seconds()
+var timeDay = 24 * time.Hour.Seconds()
+var timeWeek = 7 * 24 * time.Hour.Seconds()
+var timeYear = 365 * timeDay
+
+func parseRetentionTimeString(duration string) (int64, error) {
```

Comment:
This should already have in util package as well.

Reply:
I don't found this method, could you point this method?